test/_init_tests.js

```diff
@@ -291,60 +269,45 @@ function warnForConsoleError() {
             'error> });';
     // TODO(rsimha, #14432): Throw an error here after all tests are fixed.
     originalConsoleError(errorMessage + '\'\n' + helpMessage);
-  });
+  }
+
+  const stub = consoleSandbox.stub(console, 'error').callsFake(record);
   this.allowConsoleError = function(func) {
-    dontWarnForConsoleError();
+    stub.reset();
+    stub.callsFake(() => {});
     func();
+
     try {
-      expect(consoleErrorStub).to.have.been.called;
+      expect(stub).to.have.been.called;
     } catch (e) {
       const helpMessage =
           'The test "' + testName + '" contains an "allowConsoleError" block ' +
           'that didn\'t result in a call to console.error.';
       // TODO(rsimha, #14432): Throw an error here after all tests are fixed.
       originalConsoleError(helpMessage);
+    } finally {
+      stub.callsFake(record);
     }
-    warnForConsoleError();
   };
 }
 
-// Used during sections of tests where an error is expected.
-function dontWarnForConsoleError() {
-  if (consoleErrorSandbox) {
-    consoleErrorSandbox.restore();
-  }
-  consoleErrorSandbox = sinon.sandbox.create();
-  consoleErrorStub =
-      consoleErrorSandbox.stub(console, 'error').callsFake(() => {});
-}
-
-// Used to restore error level logging after each test.
-function restoreConsoleError() {
-  consoleErrorSandbox.restore();
-}
-
 // Used to silence info, log, and warn level logging during each test.
 function stubConsoleInfoLogWarn() {
-  if (consoleInfoLogWarnSandbox) {
-    consoleInfoLogWarnSandbox.restore();
-  }
-  consoleInfoLogWarnSandbox = sinon.sandbox.create();
-  consoleInfoLogWarnSandbox.stub(console, 'info').callsFake(() => {});
-  consoleInfoLogWarnSandbox.stub(console, 'log').callsFake(() => {});
-  consoleInfoLogWarnSandbox.stub(console, 'warn').callsFake(() => {});
+  consoleSandbox = sinon.createSandbox();
+  consoleSandbox.stub(console, 'info').callsFake(() => {});
+  consoleSandbox.stub(console, 'log').callsFake(() => {});
+  consoleSandbox.stub(console, 'warn').callsFake(() => {});
 }
 
-// Used to restore info, log, and warn level logging after each test.
-function restoreConsoleInfoLogWarn() {
-  consoleInfoLogWarnSandbox.restore();
+if (!sinon.sandbox) {
+  sinon.sandbox = sinon.createSandbox();
```

Comment:
Move this into a `before()` block and do whatever clean up is necessary (like calling `restore()`?) in `after()`.